uber/custom_tags.py

```diff
@@ -1,74 +1,136 @@
+"""
+Portions of this file are copied from the Django project. Relevant portions are
+Copyright (c) Django Software Foundation and individual contributors.
+Licensed under BSD license, see full license for details:
+https://github.com/django/django/blob/4696078832f486ba63f0783a0795294b3d80d862/LICENSE
+"""
+
 from uber.common import *
 
 
-@register.filter
-def datetime(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
-    return ' '.join(dt.astimezone(c.EVENT_TIMEZONE).strftime(fmt).split()).replace('AM', 'am').replace('PM', 'pm')
+def safe_string(str):
+    return jinja2.Markup(str)
+
 
-from datetime import datetime  # noqa: now that we've registered our filter, re-import the "datetime" class to avoid conflicts
+@JinjaEnv.jinja_filter(name='date')
+def date_filter(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
+    # TODO: if this is a datetime.time object, do we have to correct for timezone with .astimezone()?
+    return ' '.join(dt.strftime(fmt).split()).replace('AM', 'am').replace('PM', 'pm') if dt else ''
```

Comment:
Quick question about the `shift_end` filter... I see a lot of code like this:
```
start : "{{ job.start_time_local|datetime("%Y-%m-%dT%H:%M:%S") }}",
end : "{{job.start_time_local|shift_end(job.duration) }}",
```

However the `Job` class has an `end_time` property that is doing almost the same thing as the `shift_end` filter. Could we use the `end_time` property instead, get rid of the `shift_end` filter entirely, and do something like:
```
start : "{{ job.start_time_local|datetime("%Y-%m-%dT%H:%M:%S") }}",
end : "{{job.end_time_local|datetime("%Y-%m-%dT%H:%M:%S") }}",
```